lib/plugins/aws/deploy/compile/events/sns/index.js

```diff
@@ -26,22 +26,65 @@ class AwsCompileSNSEvents {
             let displayName = '';
 
             if (typeof event.sns === 'object') {
-              ['topicName', 'displayName'].forEach((property) => {
-                if (typeof event.sns[property] === 'string') {
-                  return;
+              if (event.sns.topicArn) {
```

Comment:
this should be renamed from `topicArn` to `arn` so that it follows the same pattern like the `stream` event (https://serverless.com/framework/docs/providers/aws/events/streams/)